src/main/java/com/exasol/projectkeeper/validators/dependencies/MavenDependencyReader.java

```diff
@@ -0,0 +1,119 @@
+package com.exasol.projectkeeper.validators.dependencies;
+
+import static com.exasol.projectkeeper.validators.dependencies.Dependency.Scope.*;
+
+import java.io.File;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.maven.model.*;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.project.ProjectBuildingException;
+
+import com.exasol.errorreporting.ExaError;
+import com.exasol.projectkeeper.pom.MavenArtifactModelReader;
+import com.exasol.projectkeeper.pom.MavenFileModelReader;
+import com.exasol.projectkeeper.validators.dependencies.Dependency.Scope;
+
+/**
+ * This class reads all dependencies of a pom file (including the plugins) together with their license.
+ */
+public class MavenDependencyReader {
+    private final MavenFileModelReader fileModelReader;
+    private final MavenArtifactModelReader artifactModelReader;
+
+    /**
+     * Create a new instance of {@link MavenDependencyReader}.
+     * 
+     * @param fileModelReader     pom file parser
+     * @param artifactModelReader maven dependency reader
+     */
+    public MavenDependencyReader(final MavenFileModelReader fileModelReader,
+            final MavenArtifactModelReader artifactModelReader) {
+        this.fileModelReader = fileModelReader;
+        this.artifactModelReader = artifactModelReader;
+    }
+
+    /**
+     * Read the dependencies of the pom file (including plugins).
+     * 
+     * @param pomFile pom file to read
+     * @return list of dependencies
+     */
+    public List<com.exasol.projectkeeper.validators.dependencies.Dependency> readDependencies(final File pomFile) {
+        final Model model = parsePomFile(pomFile);
+        return getDependenciesIncludingPlugins(model).map(this::getDependenciesLicense).collect(Collectors.toList());
+    }
+
+    private Stream<Dependency> getDependenciesIncludingPlugins(final Model model) {
+        return Stream.concat(model.getDependencies().stream(),
+                model.getBuild().getPlugins().stream().map(this::convertPluginToDependency));
+    }
+
+    private Dependency convertPluginToDependency(final Plugin plugin) {
+        final Dependency dependency = new Dependency();
+        dependency.setGroupId(plugin.getGroupId());
+        dependency.setArtifactId(plugin.getArtifactId());
+        dependency.setVersion(plugin.getVersion());
+        dependency.setScope("plugin");
+        return dependency;
+    }
+
+    private com.exasol.projectkeeper.validators.dependencies.Dependency getDependenciesLicense(
```

Comment:
This full class name is not reader-friendly. Maybe we can rename the class to avoid these name conflicts?

Reply:
you're right